### PR TITLE
8264791: java/util/Random/RandomTestBsi1999.java failed "java.security.SecureRandom nextFloat consecutive"

### PR DESCRIPTION
**Pull request by JimLaskey**
SecureRandom is stochastic and will become unpredictable over time (hence the intermittent nature).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264791](https://bugs.openjdk.java.net/browse/JDK-8264791): java/util/Random/RandomTestBsi1999.java failed "java.security.SecureRandom nextFloat consecutive"

### Reviewers
 * [Joe Darcy](https://openjdk.java.net/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to 5f39180eb961f03c87b895085ee56afeb4d54ad5
 * [Brian Burkhalter](https://openjdk.java.net/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to 5f39180eb961f03c87b895085ee56afeb4d54ad5
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**) ⚠️ Review applies to 5f39180eb961f03c87b895085ee56afeb4d54ad5

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3359/head:pull/3359` \
`$ git checkout pull/3359`

Update a local copy of the PR: \
`$ git checkout pull/3359` \
`$ git pull https://git.openjdk.java.net/jdk pull/3359/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3359`

View PR using the GUI difftool: \
`$ git pr show -t 3359`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3359.diff">https://git.openjdk.java.net/jdk/pull/3359.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/util/Random/RandomTestBsi1999.java [hunk 1]
```diff
@@ -439,7 +439,10 @@ public static void main(String[] args) {
         RandomGeneratorFactory.all().forEach(factory -> {
             setRNG(factory.name());
 
-            if (factory.name().equals("Random")) {
+            if (currentRNG.equals("SecureRandom")) {
+                // Bug: 8264791
```
Comment (jddarcy) [inline test/jdk/java/util/Random/RandomTestBsi1999.java:443]: It is generally not necessary to include bug ids in the text of changesets.

File: test/jdk/java/util/Random/RandomTestBsi1999.java [hunk 2]
```diff
@@ -439,7 +439,10 @@ public static void main(String[] args) {
         RandomGeneratorFactory.all().forEach(factory -> {
             setRNG(factory.name());
 
-            if (factory.name().equals("Random")) {
```
Comment (jddarcy) [inline test/jdk/java/util/Random/RandomTestBsi1999.java:442]: A string-switch on the currentRNG name might be better structured here.

**Review by jddarcy — APPROVED** [review]

**Review by bplb — APPROVED** [review]

File: test/jdk/java/util/Random/RandomTestBsi1999.java (hunk 1 above)
Comment (dcubed-ojdk) [reply test/jdk/java/util/Random/RandomTestBsi1999.java:443]: Should you add an "@bug 8264791" in the header instead?

**Review by dcubed-ojdk — APPROVED** [review]

**Pull request closed without merging**